### ISSUE TITLE
Add ability to dynamically set fields in the Summary section of the Dataset Page

### ISSUE THREAD

**Issue by scolapasta** [IQSS/dataverse#4059]
Per installation; set up as setting - 
some fields may have special formatting, as now, since they will be from Citation block; other fields would be displayed generically

**Comment by pdurbin** [comment]
@scolapasta and I discussed this issue briefly after standup.

The setting will be installation-wide and will correspond to a number of `DatasetFieldType` objects. I assume we'll want to control the display order as well.

I should look out and preserve specific display logic "above the fold" on fields such as "keyword", "description" and others.

**Comment by pdurbin** [comment]
I pushed a new branch called `4059-dynamic-fields` to add a new setting to allow @pameyer and others to list, in order, the field they want "above the fold" on the dataset page. I'll paste a screenshot below of some random fields I picked but I asked @pameyer to provide his custom metadata block, the Solr schema he uses, and a JSON file to create one of his datasets (you're welcome to simply attach them to this issue).

I asked @rbhatta99 to look at the two FIXMEs I put in the code, especially the one about preserving the order of the custom fields.

I asked @mheppler to take a look the the copy and paste I did from metadataFragment.xhtml

@scolapasta said he can advise on dataset field stuff in general.

![screen shot 2017-09-14 at 12 08 44 pm](https://user-images.githubusercontent.com/21006/30440804-970a1586-9945-11e7-8eb2-585a220550bf.png)
